buildSrc/src/main/java/UpdateContributorStatisticsTask.kt

```diff
@@ -0,0 +1,173 @@
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import java.net.URL
+
+/** Counts the changes made by each author from the git log, merges multiple authors that are linked
+ *  to the same github account and writes the result into a JSON file. */
+open class UpdateContributorStatisticsTask : DefaultTask() {
+    @get:Input lateinit var targetFile: String
+    @get:Input var skipCommits: Set<String> = setOf()
+    @get:Input var skipCommitRegex: Regex? = null
+    @get:Input lateinit var codeFileRegex: Regex
+    @get:Input lateinit var assetFileRegex: Regex
+    @get:Input lateinit var interfaceMarkupRegex: Regex
+    @get:Input lateinit var githubApiToken: String
+
+    @TaskAction fun run() {
+        val contributors = getContributors()
+
+        for (contributor in contributors) {
+            val githubUser = downloadGithubUserDetails(contributor.hash!!)
+            contributor.githubUsername = githubUser?.login
+        }
+
+        val mergedContributors = mergeContributorsByGithubAccount(contributors)
+            .filter { it.linesOfCodeChanged + it.assetFilesChanged + it.linesOfInterfaceMarkupChanged > 0 }
+
+        writeContributors(mergedContributors.sortedByDescending { it.linesOfCodeChanged })
+    }
+
+    private fun getContributors(): Collection<Contributor> {
+        val result = mutableMapOf<String, Contributor>()
+        val assetFilesChangedByCommit = mutableMapOf<String, Int>()
+        val linesOfInterfaceMarkupChangedByCommit = mutableMapOf<String, Int>()
+        val linesOfCodeChangedByCommit = mutableMapOf<String, Int>()
+        val ignoredCommits = mutableSetOf<String>()
+        val command = "git log --pretty=\"name: %an%nhash: %H%nsubject: %s%ntimestamp: %at\" --numstat"
+        Runtime.getRuntime().exec(command).inputStream.bufferedReader().useLines { lines ->
```

Comment:
For me (on Ubuntu Linux), the task fails because the output of this process is empty (i.e. `lines` is an empty sequence).

I found out that this works:

```suggestion
        ProcessBuilder(
            "git",
            "log",
            "--pretty=name: %an%nhash: %H%nsubject: %s%ntimestamp: %at",
            "--numstat",
        ).start().inputStream.bufferedReader().useLines { lines ->
```

Note that there are no single or double quotes for the `pretty` parameter, adding them (i.e. `"--pretty=\"…\""` or `"--pretty='…'"`) will make them appear in the output as well.

Does this work for you on Windows, too, @westnordost?

Reply:
It does, thanks!